### PR TITLE
Use batch_input_shape in input layer even if input_tensor is set

### PR DESCRIPTION
**Pull request by smistad**
If an input layer is supplied an input_tensor among its arguments, it will try to determine the batch_input_shape based on this tensor. 

I am using a separate pre-trained network inside a keras custom loss function, and sending the y_pred and y_true through a network. This resulted in the shape (None, None, None, None), which eventually leads to the program failing. I know the shape beforehand, however, setting the batch_input_shape didn't help. In this PR, I made it so that the input layer will always use batch_input_shape if it is set.

### CODE REVIEW

**Comment by fchollet** [comment]
That's reasonable. However do not that you can set a tensor shape information manually with TF, via `x.set_shape(shape)`.
